File: 6-paradigmas/1029.c
#include<stdio.h>
//
int call=0,repetion=0;
int fibonacci(int x){
	if(x==1|!x){
		(x)?++call:call;
		return 1;
	}
	++repetion;
	return fibonacci(x-1)+fibonacci(x-2);
}
//
int main(){
	int r;
	for(scanf("%i",&r);r>0;--r){
		int fi;
		scanf("%i",&fi);
		if(fi==1){
			printf("fib(1) = 1 calls = 1\n");
		}else{
			fibonacci(fi);
			printf("fib(%i) = %i calls = %i\n",fi,repetion*2,call);
		}
		repetion=call=0;
	}
	return 0;
}

File: 6-paradigmas/1029.py
repetion,calls=[0]*40,[0]*40
repetion[0],calls[0]=0,1
repetion[1],calls[1]=1,1
for i in range(2,39):
    repetion[i]=repetion[i-1]+repetion[i-2]
    calls[i]=calls[i-1]+calls[i-2]+1
for i in range(int(input())):
    x=int(input())
    print('fib({}) = {} calls = {}'.format(x,calls[x]-1,repetion[x]))
